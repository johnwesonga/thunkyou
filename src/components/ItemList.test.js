import React from 'react';
import {ItemList} from './ItemList';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';


describe('ItemList Component', () => {
    const initialState = {items: {}}
    const mockStore = configureStore()
    let store,container, wrapper

    const mockedCallback = () => Promise.resolve({foo: 'bar'});
    let promise;
    const loadData = () => {
      promise = Promise.resolve().then(mockedCallback);
      return promise;
    };

    const itemListProps = {
        fetchData: jest.fn(),
        items: {},
        hasErrored: false,
        isLoading: false,
    }

    beforeEach(()=>{
       store = mockStore(initialState)
       wrapper = shallow(<Provider store={store}><ItemList {...itemListProps}/></Provider> )  
    });

    it(' snapshot ItemList ', () =>{
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('render the ItemList component', () => {
        expect(wrapper.find(ItemList).length).toEqual(1)
    });

    it('+++ check Prop matches with initialState', () => {
        expect(wrapper.find(ItemList).prop('items')).toEqual(initialState.items)
     });

});
