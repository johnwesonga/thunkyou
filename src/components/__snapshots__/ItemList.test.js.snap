// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList Component  snapshot ItemList  1`] = `
<ItemList
  fetchData={[Function]}
  hasErrored={false}
  isLoading={false}
  items={Object {}}
/>
`;
